shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -0,0 +1,723 @@
+        for handler in handlers:
+            response = handler(*args, **kwargs)
+            if response:
+                return response
+
+    # Copied from manual_product_transfer
+    def _find_location_move_lines_domain(self, location, product, lot=None):
+        domain = [
+            ("location_id", "=", location.id),
+            ("product_id", "=", product.id),
+            ("state", "in", ("assigned", "partially_available")),
+            ("picking_id.user_id", "in", (False, self.env.uid)),
+        ]
+        if lot:
+            domain = AND([domain, [("lot_id", "=", lot.id)]])
+        return domain
+
+    # Copied from manual_product_transfer
+    def _find_location_move_lines(self, location, product, lot=None):
+        """Find existing move lines in progress related to the source location
+        but not linked to any user.
+        """
+        domain = self._find_location_move_lines_domain(location, product, lot=lot)
+        return self.env["stock.move.line"].search(domain)
+
+    # Copied from manual_product_transfer
+    def _create_move_from_location(
+        self, location, product, quantity, lot=None, packaging=None
+    ):
+        picking_type = self.picking_types
+        move_vals = {
+            "name": product.name,
+            "company_id": picking_type.company_id.id,
+            "product_id": product.id,
+            "product_uom": product.uom_id.id,
+            "product_uom_qty": quantity,
+            "location_id": location.id,
+            "location_dest_id": picking_type.default_location_dest_id.id,
+            "origin": self.work.menu.name,
+            "picking_type_id": picking_type.id,
+        }
+        move = self.env["stock.move"].create(move_vals)
+        move._action_confirm(merge=False)
+        move.with_context(
+            {"force_reservation": self.work.menu.allow_force_reservation}
+        )._action_assign()
+        assert move.state == "assigned", "The reservation of quantities has failed"
+        move_line = move.move_line_ids
+        stock = self._actions_for("stock")
+        if self.work.menu.no_prefill_qty:
+            # We ensure the qty_done is 0 here, so we can set it manually after
+            # to avoid the split of the move line by 'mark_move_line_as_picked'.
+            stock.mark_move_line_as_picked(move_line, quantity=0)
+            # Set the initial qty_done to 1 for product and lot
+            qty_done = 1
+            if packaging:
+                qty_done = packaging.qty
+            move_line.qty_done = qty_done
+        else:
+            if packaging:
+                # We ensure the qty_done is 0 here, so we can set it manually after
+                # to avoid the split of the move line by 'mark_move_line_as_picked'.
+                stock.mark_move_line_as_picked(move_line, quantity=0)
+                move_line.qty_done = packaging.qty
+            else:
+                stock.mark_move_line_as_picked(move_line)
```

Comment:
As you just created the move for the quantity to process, I think you can always call 
```suggestion
        stock.mark_move_line_as_picked(move_line)
```

Reply:
does it mean we have to ignore no_prefill_qty and always pick what is available?

Reply:
at second read, what you did looks right. Just in case of product packaging, the quantity todo looks wrong.

This comment can be resolved